### ISSUE TITLE
Invis and blur effects don't work with AA on some graphics cards.

### ISSUE THREAD

**Issue by skittles1** [OpenMeridian105/Meridian59#250]
Recently the classic client's graphics were updated with anti-aliasing and mipmaps. However, a small minority of graphics cards can't draw the two effects (invis and blur) that both require drawing everything rendered so far to a buffer and modifying it (to my knowledge, this is what is being done).

As far as I can tell, we can't selectively turn AA off for these two effects, and it's not clear if something is actually wrong with the code in the [D3DRenderFramebufferTextureCreate function](https://github.com/OpenMeridian105/Meridian59/blob/develop/clientd3d/d3drender.c#L9095) or if those cards just aren't handling this edge case correctly. Unfortunately I don't have a system with the broken effects to do live debugging on, so my options are a bit limited.

@cyberjunk if you get some spare time to help with this it would be greatly appreciated.

**Comment by cyberjunk** [comment]
I just tested it and it is still broken for me...
However this [StretchRect](https://github.com/OpenMeridian105/Meridian59/blob/develop/clientd3d/d3drender.c#L9121) call does NOT return any error, just OK.

Removing the AA block [here](https://github.com/OpenMeridian105/Meridian59/blob/develop/clientd3d/d3ddriver.c#L118) makes invis work again as expected.

I noticed one interesting thing... I think the [D3DPRESENTFLAG_LOCKABLE_BACKBUFFER](https://github.com/OpenMeridian105/Meridian59/blob/develop/clientd3d/d3ddriver.c#L114) flag should actually be set (given what the code in D3DRenderFramebufferTextureCreate() does. I can set this flag without issues, as long as I keep the AA block disabled. However, once I enable the AA block and also set the flag, then the device creation [here](https://github.com/OpenMeridian105/Meridian59/blob/develop/clientd3d/d3ddriver.c#L130) fails.

Does this happen for you as well ?

**Comment by skittles1** [comment]
Yeah I get the same behavior if I enable that flag - no device creation with AA block enabled, works with AA commented out.

**Comment by cyberjunk** [comment]
Ok. Let's see...

[D3DPRESENTFLAG_LOCKABLE_BACKBUFFER](https://msdn.microsoft.com/en-us/library/windows/desktop/bb172586(v=vs.85).aspx) very likely only refers to the capability to transfer data between GPU backbuffer and system memory (both directions).

That is why [GetRenderTargetData](https://msdn.microsoft.com/en-us/library/windows/desktop/bb174405(v=vs.85).aspx) (GPU->SYS) says:
>> This method will fail if:
* The render target is multisampled.

and [UpdateSurface](https://msdn.microsoft.com/en-us/library/windows/desktop/bb205857(v=vs.85).aspx) (SYS->GPU) says:

>> This function has the following restrictions.
* Neither surface can be created with multisampling. The only valid flag for both surfaces is D3DMULTISAMPLE_NONE.

Now, in our case we (luckily) don't want to transfer data between GPU and SYS memory, but instead from GPU to GPU using [StretchRect](https://msdn.microsoft.com/en-us/library/windows/desktop/bb174471(v=vs.85).aspx). That is why we did not need the [D3DPRESENTFLAG_LOCKABLE_BACKBUFFER](https://msdn.microsoft.com/en-us/library/windows/desktop/bb172586(v=vs.85).aspx) so far to make it work (and don't need it all).
